diff --git a/node_modules/cordova-plugin-serial/lib/usbseriallibrary.jar b/node_modules/cordova-plugin-serial/lib/usbseriallibrary.jar
deleted file mode 100644
index 3f9e2df..0000000
Binary files a/node_modules/cordova-plugin-serial/lib/usbseriallibrary.jar and /dev/null differ
diff --git a/node_modules/cordova-plugin-serial/plugin.xml b/node_modules/cordova-plugin-serial/plugin.xml
index 2bc7f70..b81126e 100644
--- a/node_modules/cordova-plugin-serial/plugin.xml
+++ b/node_modules/cordova-plugin-serial/plugin.xml
@@ -22,7 +22,7 @@
         </config-file>
         <source-file src="src/android/fr/drangies/cordova/serial/Serial.java" target-dir="src/fr/drangies/cordova/serial" />
         <source-file src="src/android/fr/drangies/cordova/serial/UsbBroadcastReceiver.java" target-dir="src/fr/drangies/cordova/serial" />
-        <source-file src="lib/usbseriallibrary.jar" target-dir="libs" />
+        <framework src="com.github.mik3y:usb-serial-for-android:3.7.3" />
     </platform>
 
     <!-- ubuntu -->
diff --git a/node_modules/cordova-plugin-serial/src/android/fr/drangies/cordova/serial/Serial.java b/node_modules/cordova-plugin-serial/src/android/fr/drangies/cordova/serial/Serial.java
index a32d36f..c44d774 100644
--- a/node_modules/cordova-plugin-serial/src/android/fr/drangies/cordova/serial/Serial.java
+++ b/node_modules/cordova-plugin-serial/src/android/fr/drangies/cordova/serial/Serial.java
@@ -285,8 +285,8 @@ public class Serial extends CordovaPlugin {
                     try {
                         Log.d(TAG, data);
                         byte[] buffer = hexStringToByteArray(data);
-                        int result = port.write(buffer, 1000);
-                        callbackContext.success(result + " bytes written.");
+                        port.write(buffer, 1000);
+                        callbackContext.success(buffer.length + " bytes written.");
                     }
                     catch (IOException e) {
                         // deal with error
